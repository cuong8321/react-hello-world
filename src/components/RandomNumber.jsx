import React from 'react'
import Paper from 'material-ui/Paper'

export default class RandomNumber extends React.Component {
  constructor (props) {
    super(props)
    const {display = String, init = Math.random} = props
    const value = init()
    const displayValue = () => display(value)
    this.state = {displayValue}
  }

  render () {
    return <Paper zDepth={0}>
      <span className='label'>Random Number: </span>
      <span className='value'>{this.state.displayValue()}</span>
    </Paper>
  }
}
