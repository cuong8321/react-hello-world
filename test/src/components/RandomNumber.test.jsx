import './__lib__/material-ui-mocks.jsx'
import React from 'react'
import renderer from 'react-test-renderer'
import RandomNumber from '../../../src/components/RandomNumber.jsx'

describe('Snapshot: components/RandomNumber.jsx', () => {
  const init = () => 0.123456789

  test('with default display function', () => {
    const tree = renderer.create(<RandomNumber init={init} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with specified display function', () => {
    ; [
      value => JSON.stringify({value}),
      value => Number(value).toString(16).slice(2, 16),
      value => Number(value).toString(36).slice(2, 13),
      value => value * value,
      value => [0, 1, 2, 3].map(x => value * x).join(', ')
    ]
      .map(display => (<RandomNumber init={init} display={display} />))
      .map(element => renderer.create(element))
      .map(instance => instance.toJSON())
      .forEach(tree => expect(tree).toMatchSnapshot())
  })
})
