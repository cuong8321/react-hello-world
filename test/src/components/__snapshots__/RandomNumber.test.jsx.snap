// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: components/RandomNumber.jsx with default display function 1`] = `
<mocked-component
  zDepth={0}
>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    0.123456789
  </span>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 1`] = `
<mocked-component
  zDepth={0}
>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    {"value":0.123456789}
  </span>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 2`] = `
<mocked-component
  zDepth={0}
>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    1f9add3739635f
  </span>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 3`] = `
<mocked-component
  zDepth={0}
>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    4fzzzxjylrx
  </span>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 4`] = `
<mocked-component
  zDepth={0}
>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    0.01524157875019052
  </span>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 5`] = `
<mocked-component
  zDepth={0}
>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    0, 0.123456789, 0.246913578, 0.370370367
  </span>
</mocked-component>
`;
